common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -931,4 +931,34 @@ public static Geometry geometricMedian(Geometry geometry) throws Exception {
         return geometricMedian(geometry, DEFAULT_TOLERANCE, DEFAULT_MAX_ITER, false);
     }
 
+    public static Geometry boundingDiagonal(Geometry geometry) {
+        if (geometry.isEmpty()) {
+            return GEOMETRY_FACTORY.createLineString();
+        }else {
+            Envelope envelope = geometry.getEnvelopeInternal();
```

Comment:
getEnvelopeInternal() also traverses all coordinates I guess. Why not just have a single loop and traverse all coordinates?

Reply:
Good point, I started out with that initially but ended up using envelope for simplicity. I did not take into account its internal implementation.
Reverted to using 1 loop